code/modules/modular_computers/computers/modular_computer/ui.dm

```diff
@@ -44,13 +44,22 @@
 		data["_PC"] = headerdata
 		. = data
 
-	var/datum/computer_file/data/autorun = hard_drive.find_file_by_name("autorun")
+	var/obj/item/computer_hardware/hard_drive/composite_drive = new /obj/item/computer_hardware/hard_drive(src)
+	if(hard_drive)
+		for(var/stored_file in hard_drive.stored_files)
+			composite_drive.store_file(stored_file)
+	if(portable_drive)
+		for(var/stored_file in portable_drive.stored_files)
+			composite_drive.store_file(stored_file)
```

Comment:
This is a very hacky solution, as it involvs creating a new "composite_drive" every time the ui is updated.